swift build and bundle optimisations and fixes

Akemi opened the pull request:
includes #6961. tested this on a machine that had an older than swift 5 version where ABI compatibility wasn't yet there and it just crashed.

jeeb commented on wscript_build.py:
```diff
@@ -580,12 +580,16 @@ def swift(task):
             source       = main_fn_c,
             use          = ctx.dependencies_use() + ['objects'],
-            add_object   = "osdep/macOS_swift.o",
+            add_object   = additonal_objects,
```
Nit: using plural (object**s**), but content is a single string. See if this works with a list and then just append() to it in the if check in case it works.

jeeb commented on TOOLS/dylib-unhell.py:
```diff
@@ -89,8 +89,16 @@ def process_libraries(libs_dict, binary, processed = []):
 def process_swift_libraries(binary):
     command = ['xcrun', '--find', 'swift-stdlib-tool']
     swiftStdlibTool = subprocess.check_output(command, universal_newlines=True).strip()
+    #from xcode11 the dynamic swift libs reside in a seperate directory from
```
`separate`. Also I think the Python comment style has a space after the hash symbol?

jeeb commented on TOOLS/dylib-unhell.py:
```diff
@@ -28,6 +28,11 @@ def otool(objfile):
     return set(filter(partial(is_user_lib, objfile), output.split()))
 
+def get_rpaths_dev_tools(binary):
+    command = "otool -l %s | grep -A2 LC_RPATH | grep path | grep \"Xcode\|CommandLineTools\" | awk '{ print $2 }'" % binary
```
I mean, the command right on top of this one in `otool()` has the same issue, so I won't require this to be changed here (I'd rather both get fixed at the same time at some point if needed), but I'm pretty sure this would start failing due to lack of quoting with the `%s` in case spaces in paths? Possibly also the awk might fail in that case, but I don't recall which part of the output $2 was again :) .

jeeb commented on TOOLS/dylib-unhell.py:
```diff
@@ -116,6 +128,9 @@ def main():
     while libs_processed is not None:
         libs_processed = process_libraries(libs, binary, libs_processed)
 
+    print(">> removing rapths of dev tools ")
```
`rapths` → `rpath` (commit message as well) and probably "removing rpath definitions towards dev. tools" or something?

jeeb commented:
Once again, thanks for taking care of handling the very flux state in which Apple seems to be keeping the Swift toolchains.

Akemi commented:
i addressed your concerns besides the awk in the old otool function, don't want to break that now.

also tested this with the released xcode 11 version and nothing changed from the Betas. so these commits should be good to go.

Akemi commented:
merged as:
4a2f239842762dc11bb67a5cd54ab2b6f2fb4003
9e52d369626483a7523ab5800cd5443db0cfb576
e3972746dc18473ad577d04547d0f05775da4658
f64c0115ae475c8584e0825340b7488be4095c2c
